PubHub/PubHub.BookMobile/Auth/User.cs

```diff
@@ -58,16 +59,52 @@ internal static void Unset()
             SecureStorage.Default.RemoveAll();
         }
 
+        /// <summary>
+        /// Checks the authentication state of the current user and if their claim is goint to expire in 10 minutes or less, tries to refresh their token pair
+        /// </summary>
+        /// <param name="authService"></param>
+        /// <returns>The <see cref="Task"/> that represents the <see langword="asynchronous"/> operation</returns>
+        internal static async Task CheckStateAndTryRefreshAsync(IAuthenticationService authService)
+        {
+            (bool isSuccess, TokenInfo? tokens) = await TryGetCachedTokenAsync();
+
+            if (isSuccess)
+            {
+                var handler = new JwtSecurityTokenHandler();
+                var token = handler.ReadJwtToken(tokens?.Token);
+                var utcExpire = new DateTimeOffset(token.ValidTo);
+                var expireDate = utcExpire.DateTime;
+
+                if (expireDate < DateTime.UtcNow.AddMinutes(10))
+                {
+                    var result = await authService.RefreshTokenAsync();
+
+                    if (!result.IsSuccess || result.Instance is null)
+                    {
+                        if (result.StatusCode == HttpStatusCode.Unauthorized)
+                            await Shell.Current.CurrentPage.DisplayAlert(UnauthorizedError.TITLE, UnauthorizedError.ERROR_MESSAGE, UnauthorizedError.BUTTON_TEXT);
+                        else
+                            await Shell.Current.CurrentPage.DisplayAlert(NoConnectionError.TITLE, NoConnectionError.ERROR_MESSAGE, NoConnectionError.BUTTON_TEXT);
+
+                        Unset();
+                        return;
+                    }
+
+                    await SetAsync(result.Instance);
+                }
+            }
+        }
```

Comment:
Did you get this to work? :O

Reply:
That methods works, yes, but I have nowhere to execute it :D